kafka/source/pkg/adapter/adapter.go

```
@@ -152,3 +154,12 @@ func (a *Adapter) Handle(ctx context.Context, msg *sarama.ConsumerMessage) (bool
 	_ = a.reporter.ReportEventCount(reportArgs, res.StatusCode)
 	return true, nil
 }
+
+func safeCloseHTTPResponseBody(response *http.Response, adapter *Adapter) {
+	if response != nil && response.Body != nil {
+		err := response.Body.Close()
```

IMO we should discard this error